assignments/arrays.js

```diff
@@ -63,32 +63,107 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// {"id":33,"car_make":"Jeep","car_model":"Wrangler","car_year":2011},
+console.log(`Car 33 is a ` + inventory[32].car_year + ` ` + inventory[32].car_make + ` ` + inventory[32].car_model );
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+// {"id":50,"car_make":"Lincoln","car_model":"Town Car","car_year":1999}];
+let lastCar = inventory.length-1;
+console.log(`Make and model of the last car is ` + inventory[lastCar].car_make +` and `+ inventory[lastCar].car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+function sortByCarModel(){
+    let extrCar="";
+    let left = 0;
 
+    let carModels = [];
+    for (let i=0; i< inventory.length-1; i++) {
+        carModels[i] = inventory[i].car_model;
+    }
+    let right = carModels.length-1;
+
+    for (let i=0; i<carModels.length;i++) {
+
+        for (let i = left; i < right; i++) {
+            if (carModels[i].charAt(0) > carModels[i + 1].charAt(0)) {
+                extrCar = carModels[i];
+                carModels[i] = carModels[i + 1];
+                carModels[i + 1] = extrCar;
+            }
+        }
+        right--;
+        for (let j = right; j > left; j--) {
+            if (carModels[right].charAt(0) < carModels[right - 1].charAt(0)) {
+                extrCar = carModels[right];
+                carModels[right] = carModels[right - 1];
+                carModels[right - 1] = extrCar;
+                left++;
+            }
+        }
+    }
+    return carModels;
+}
+console.log(sortByCarModel());
+// OR
+function easySort(){
+    let carModels = [];
+    for (let i=0; i< inventory.length-1; i++) {
+        carModels[i] = inventory[i].car_model;
+    }
+    return "Second way to sort cars by models: " +carModels.sort();
+}
+console.log(easySort());
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+function carYears(){
+    let carYears = [];
+    for (let i=0; i<inventory.length; i++){
+            carYears[i] = inventory[i].car_year;
+    }
+    return  carYears;
```

Comment:
or you can console.log(carYears) right here instead of the return, although I like what you did here as it is generally a good idea to get in the habit of returning something from a function.